src/services/poolTogether.js

```diff
@@ -0,0 +1,416 @@
+  if (gasToken) {
+    transactionPayload = { ...transactionPayload, gasToken };
+  }
+
+  transactionPayload = { ...transactionPayload, txFeeInWei };
+
+  return {
+    gasToken,
+    txFeeInWei,
+    transactionPayload,
+  };
+}
+
+export const checkPoolAllowance = async (symbol: string, address: string): Promise<boolean> => {
+  const {
+    poolContractAddress,
+    tokenContractAddress: contractAddress,
+    tokenContract: contract,
+  } = getPoolTogetherTokenContract(symbol);
+  let hasAllowance = false;
+  try {
+    const allowanceResult = await contract.allowance(address, poolContractAddress);
+    if (allowanceResult) {
+      hasAllowance = allowanceResult.toString() !== '0';
+    }
+  } catch (e) {
+    reportLog('Error checking PoolTogether Allowance', {
+      address,
+      poolContractAddress,
+      contractAddress,
+      symbol,
+      message: e.message,
+    }, Sentry.Severity.Error);
+    hasAllowance = false;
+  }
+  return hasAllowance;
+};
+
+export async function getPurchaseTicketFeeAndTransaction(depositAmount: number, symbol: string, useGasToken: boolean) {
+  const {
+    poolAbi,
+    unitType: decimals,
+    poolContractAddress,
+  } = getPoolTogetherTokenContract(symbol);
+  const depositMethod = poolAbi.find(item => item.name === 'depositPool');
+  const valueToDeposit = utils.parseUnits(depositAmount.toString(), decimals);
+  const data = abi.encodeMethod(depositMethod, [valueToDeposit]);
+  let transactionPayload = {
+    amount: 0,
+    to: poolContractAddress,
+    symbol,
+    poolContractAddress,
+    decimals,
+    data,
+    extra: {
+      symbol,
+      amount: valueToDeposit,
+      decimals,
+    },
+    tag: POOLTOGETHER_DEPOSIT_TRANSACTION,
+  };
+
+  const { fee: txFeeInWei, gasToken } = await getSmartWalletTxFee(transactionPayload, useGasToken);
+  if (gasToken) {
+    transactionPayload = { ...transactionPayload, gasToken };
+  }
+
+  transactionPayload = { ...transactionPayload, txFeeInWei };
+
+  return {
+    gasToken,
+    txFeeInWei,
+    transactionPayload,
+  };
+}
+
+export async function getWithdrawTicketFeeAndTransaction(withdrawAmount: number, symbol: string, useGasToken: boolean) {
+  const {
+    poolAbi,
+    unitType: decimals,
+    poolContractAddress,
+  } = getPoolTogetherTokenContract(symbol);
+  const withdrawMethod = poolAbi.find(item => item.name === 'withdraw');
+  const valueToWithdraw = utils.parseUnits(withdrawAmount.toString(), decimals);
+  const data = abi.encodeMethod(withdrawMethod, [valueToWithdraw]);
+  let transactionPayload = {
+    amount: 0,
+    to: poolContractAddress,
+    symbol,
+    poolContractAddress,
+    decimals,
+    data,
+    extra: {
+      symbol,
+      amount: valueToWithdraw,
+      decimals,
+    },
+    tag: POOLTOGETHER_WITHDRAW_TRANSACTION,
+  };
+
+  const { fee: txFeeInWei, gasToken } = await getSmartWalletTxFee(transactionPayload, useGasToken);
```

Comment:
Same as above, what if we have. an error here? Is this handled in `getSmartWalletTxFee`?

Reply:
Error will be thrown and the users will need to recalculate by changing values 4bc3f3ebe83817ec7cb4f61dae81f832b3067407

Reply:
Thanks for the explanation! Just attempting to understand how events bubble up to the UI.